### PR TITLE
Refactor xtask

### PR DESCRIPTION
**Pull request by iamricks**
First time contributing to Burn and a bit new to Rust, any feedback is appreciated  🙏 

### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs
https://github.com/tracel-ai/burn/issues/1191

### Changes
Refactored using some of the code in **xtask/src/utils/cargo.rs**

### Testing

Ran `./run-checks all`

### CODE REVIEW

**Review by Luni-4 — APPROVED** [review]
Thanks a lot for your refactor! It's fine for me!

File: xtask/src/publish.rs [hunk 1]
```diff
@@ -58,74 +59,61 @@ fn remote_version(crate_name: &str) -> Option<String> {
...
-    // Run cargo publish --dry-run
-    cargo_publish(&["-p", &crate_name, "--dry-run"]);
+    // Perform dry-run to ensure everything is good for publishing
+    let dry_run_params = Params::from(["publish", "-p", &crate_name, "--dry-run"]);
+
+    run_cargo(
+        "publish",
+        dry_run_params,
+        HashMap::new(),
+        "Failed to run cargo publish --dry-run",
```
Comment (syl20bnr) [inline xtask/src/publish.rs:70]: This refactor is a good occasion to update the update message of this file, see the guidelines here https://github.com/tracel-ai/burn/blob/main/CONTRIBUTING.md#writing-expect-messages

File: xtask/src/publish.rs [hunk 2]
```diff
@@ -58,74 +59,61 @@ fn remote_version(crate_name: &str) -> Option<String> {
...
+        "--token",
+        &crates_io_token,
+    ]);
+
+    // Actually publish the crate
+    run_cargo(
+        "publish",
+        publish_params,
+        envs,
+        "Failed to publish the crate",
```
Comment (syl20bnr) [inline xtask/src/publish.rs:89]: Same here.

**Review by syl20bnr — CHANGES_REQUESTED** [review]
Thank you for this first PR !
We can use this refactor to update the expect messages at the same time to be more conventional.

File: xtask/src/publish.rs (hunk 1 above)
Comment (iamricks) [reply xtask/src/publish.rs]: Got it! I just changed them, let me know if you think they're okay 👍

**Review by syl20bnr — APPROVED** [review]
Sorry for the late review. All good 👍🏻

**Pull request merged**